Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -148,6 +171,372 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+
+end WalkingParallelPair
+
+end Limits
+
+namespace Limits
+
+open WalkingReflexivePair
+
+open WalkingReflexivePairHom
+
+variable {C : Type u} [Category.{v} C]
+
+variable {A B : C}
+
+/-- Bundle the data of a parallel pair along with a common section as a functor out of the walking
+reflexive pair -/
+def reflexivePair (f g : A ⟶ B) (s : B ⟶ A) (sl : s ≫ f = 𝟙 B) (sr : s ≫ g = 𝟙 B) :
+    (WalkingReflexivePair ⥤ C) where
+  obj x :=
+    match x with
+    | zero => B
+    | one => A
+  map h :=
+    match h with
+    | WalkingReflexivePairHom.id _ => 𝟙 _
+    | left => f
+    | right => g
+    | reflexion => s
+    | right_reflexion => g ≫ s
+    | left_reflexion => f ≫ s
+  map_comp := by
+    rintro _ _ _ ⟨⟩ g <;> cases g <;>
+    simp only [Category.id_comp, Category.comp_id, Category.assoc, sl, sr] <;>
+    try {rfl}
+    iterate simp only [Category.id_comp, Category.comp_id, ← Category.assoc, sl, sr] <;> try {rfl}
+
+/-- (Noncomputably) bundle the data of a reflexive pair as a functor out of the walking reflexive
+pair -/
+noncomputable def ofIsReflexivePair (f g : A ⟶ B) [IsReflexivePair f g] :
+    (WalkingReflexivePair ⥤ C) :=
+  reflexivePair f g (commonSection f g) (section_comp_left f g) (section_comp_right f g)
+
+/-- The natural isomorphism between the diagram obtained by forgetting the reflexion of
+`ofIsReflexivePair f g` and the original parallel pair. -/
+noncomputable def inclusionWalkingReflexivePairOfIsReflexivePairIso
+    (f g : A ⟶ B) [IsReflexivePair f g] :
+    WalkingParallelPair.inclusionWalkingReflexivePair ⋙ (ofIsReflexivePair f g) ≅
+      parallelPair f g :=
+  diagramIsoParallelPair _
+
+end Limits
+
+namespace Limits
+
+open WalkingReflexivePair
+
+open WalkingReflexivePairHom
+
+variable {C : Type u} [Category.{v} C]
+
+@[simp]
+theorem reflexivePair_reflexion_comp_left (F : WalkingReflexivePair ⥤ C) :
+    F.map reflexion ≫ F.map left = 𝟙 F.obj zero :=
+  (F.map_id zero).symm.trans (F.map_comp reflexion left)|>.symm
+
+@[simp]
+theorem reflexivePair_reflexion_comp_right (F : WalkingReflexivePair ⥤ C) :
+    F.map reflexion ≫ F.map right = 𝟙 F.obj zero :=
+  (F.map_id zero).symm.trans (F.map_comp reflexion right)|>.symm
+
+@[simp]
+theorem reflexivePair_left_comp_reflexion (F : WalkingReflexivePair ⥤ C) :
+    F.map left_reflexion = F.map left ≫ F.map reflexion := F.map_comp left reflexion
+
+@[simp]
+theorem reflexivePair_right_comp_reflexion (F : WalkingReflexivePair ⥤ C) :
+    F.map right_reflexion = F.map right ≫ F.map reflexion := F.map_comp right reflexion
+
+/-- Any functor out of `WalkingReflexivePair` is isomorphic to the reflexive pair built out of the
+images of its morphisms -/
+def diagramIsoReflexivePair (F : WalkingReflexivePair ⥤ C) :
+    F ≅ reflexivePair (F.map left) (F.map right) (F.map reflexion)
+    ((F.map_id zero).symm.trans (F.map_comp reflexion left)).symm
+    ((F.map_id zero).symm.trans (F.map_comp reflexion right)).symm :=
```

Comment:
Taking into account suggestions above, I would proceed like this:
```lean
def diagramIsoReflexivePair (F : WalkingReflexivePair ⥤ C) :
    F ≅ reflexivePair (F.map left) (F.map right) (F.map reflexion) :=
  NatIso.ofComponents (fun x ↦ match x with
    | zero => Iso.refl _
    | one => Iso.refl _)
  (fun f ↦ by cases f <;> simp)
```
(As much as possible, I feel it is slightly better not to use `eqToIso`.)
More generally, as you have defined a constructor for functors from the category `WalkingReflexivePair`, it would be nice to introduce also a constructor for natural transformations (and then natural isomorphisms) between objets in `WalkingReflexivePair ⥤ C`. Then, `diagramIsoReflexivePair` would appear as a special case of this.